cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaBrokerConfigurationBuilder.java

```diff
@@ -783,48 +804,80 @@ private void configProviders(KafkaConfiguration userConfig)    {
      *
      * @return Returns the builder instance
      */
-    public KafkaBrokerConfigurationBuilder withUserConfiguration(KafkaConfiguration userConfig, boolean injectCcMetricsReporter)  {
-        if (userConfig != null && !userConfig.getConfiguration().isEmpty()) {
-            // We have to create a copy of the configuration before we modify it
-            userConfig = new KafkaConfiguration(userConfig);
-
-            // Configure the configuration providers => we have to inject the Strimzi ones
-            configProviders(userConfig);
-
-            if (injectCcMetricsReporter)  {
-                // We configure the Cruise Control Metrics Reporter is needed
-                if (userConfig.getConfigOption(CruiseControlMetricsReporter.KAFKA_METRIC_REPORTERS_CONFIG_FIELD) != null) {
-                    if (!userConfig.getConfigOption(CruiseControlMetricsReporter.KAFKA_METRIC_REPORTERS_CONFIG_FIELD).contains(CruiseControlMetricsReporter.CRUISE_CONTROL_METRIC_REPORTER)) {
-                        userConfig.setConfigOption(CruiseControlMetricsReporter.KAFKA_METRIC_REPORTERS_CONFIG_FIELD, userConfig.getConfigOption(CruiseControlMetricsReporter.KAFKA_METRIC_REPORTERS_CONFIG_FIELD) + "," + CruiseControlMetricsReporter.CRUISE_CONTROL_METRIC_REPORTER);
-                    }
-                } else {
-                    userConfig.setConfigOption(CruiseControlMetricsReporter.KAFKA_METRIC_REPORTERS_CONFIG_FIELD, CruiseControlMetricsReporter.CRUISE_CONTROL_METRIC_REPORTER);
-                }
-            }
-
-            printSectionHeader("User provided configuration");
+    public KafkaBrokerConfigurationBuilder withUserConfiguration(KafkaConfiguration userConfig,
+                                                                 boolean injectCcMetricsReporter,
+                                                                 boolean injectStrimziMetricsReporter) {
+        // we have to create a copy of the configuration before we modify it
+        userConfig = userConfig != null
+                ? new KafkaConfiguration(userConfig)
+                : new KafkaConfiguration(reconciliation, new ArrayList<>());
+
+        // configure config.providers
+        addConfigProviders(userConfig);
+
+        // configure metric.reporters
+        maybeAddMetricReporters(userConfig, injectCcMetricsReporter, injectStrimziMetricsReporter);
+
+        // configure kafka.metrics.reporters
+        maybeAddKafkaMetricsReporters(userConfig, injectStrimziMetricsReporter);
+
+        // print user config with Strimzi injections
+        if (!userConfig.getConfiguration().isBlank()) {
+            printSectionHeader("User provided configuration with Strimzi injections");
             writer.println(userConfig.getConfiguration());
             writer.println();
+        }
+
+        return this;
+    }
+
+    private void addConfigProviders(KafkaConfiguration userConfig) {
```

Comment:
Why is this needed? Why does it duplicate the stuff from `configProviders`? It seems like this should be deleted?

Reply:
This is not a duplicate. It's part of the refactoring for the Metrics Reporter integration. This is the old configProvider. The logic is the same but using the new functions. We just extracted some common functions to improve maintenance. Do you think we shout revert it?

Reply:
The methods look pretty much the same and the old one for some reason does not seem to be used. So it is duplicate.

Reply:
PS: For the sake of history, you should keep (and if needed rename) the old method unless you have some strong reasons why to do it differently.

Reply:
@OwenCorrigan76 I think that was an oversight. The plan was to remove the old method named `configProviders`.

Reply:
@scholzj Sorry, Fede is correct. I forgot to remove the old method so it is a duplicate. The plan was to remove `configProviders` but it got back in.

Reply:
Well, as I said -> you should try to preserve the history and not delete method to replace it with the same method in the same class just in a different place. That is screwing up the diffs and the tracking of the changes.

Reply:
Duplicate removed. Apologies for the noise.